sound/soc/sof/amd/acp.c

```diff
@@ -138,25 +138,40 @@ int configure_and_run_dma(struct acp_dev_data *adata, unsigned int src_addr,
 	return ret;
 }
 
-static int psp_fw_validate(struct acp_dev_data *adata)
+static int psp_mbox_ready(struct acp_dev_data *adata)
 {
 	struct snd_sof_dev *sdev = adata->dev;
 	int timeout;
 	u32 data;
 
-	smn_write(adata->smn_dev, MP0_C2PMSG_26_REG, MBOX_ACP_SHA_DMA_COMMAND);
-
 	for (timeout = ACP_PSP_TIMEOUT_COUNTER; timeout > 0; timeout--) {
 		msleep(20);
 		smn_read(adata->smn_dev, MP0_C2PMSG_26_REG, &data);
 		if (data & MBOX_READY_MASK)
 			return 0;
 	}
 
-	dev_err(sdev->dev, "FW validation timedout: status %x\n", data & MBOX_STATUS_MASK);
+	dev_dbg(sdev->dev, "PSP error status %x\n", data & MBOX_STATUS_MASK);
 	return -ETIMEDOUT;
 }
 
+static int psp_send_cmd(struct acp_dev_data *adata)
+{
+	struct snd_sof_dev *sdev = adata->dev;
+
+	/* Check if PSP is ready for new command */
+	if (psp_mbox_ready(adata))
+		return -EBUSY;
```

Comment:
this is also not very elegant. WHy not modify psp_mbox_ready() with a bool indicating if you're checking for busy or ready status and return the appropriate error instead of modifying the returned error from the calling function